clients/src/main/java/org/apache/kafka/common/requests/MetadataRequest.java

```diff
@@ -65,6 +65,20 @@ public Builder(List<String> topics, boolean allowAutoTopicCreation) {
             this(topics, allowAutoTopicCreation, ApiKeys.METADATA.oldestVersion(),  ApiKeys.METADATA.latestVersion());
         }
 
+        public Builder(List<Uuid> topicIds) {
+            super(ApiKeys.METADATA, ApiKeys.METADATA.oldestVersion(), ApiKeys.METADATA.latestVersion());
+            MetadataRequestData data = new MetadataRequestData();
+            if (topicIds == null)
+                data.setTopics(null);
+            else {
+                topicIds.forEach(topicId -> data.topics().add(new MetadataRequestTopic().setTopicId(topicId).setName("")));
```

Comment:
Is there a reason this is setting the name to empty and not null? (or no name set at all?)

Reply:
This is old code since we do not support null names firstly, now we can remove this useless code.